NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -32,14 +19,100 @@ function Base.show(io::IO, cs::Vector{<:AbstractCategory})
   return nothing
 end
 
+Base.isless(c1::AbstractCategory, c2::AbstractCategory) = isless(label(c1), label(c2))
+
+# =================  Misc  ======================
 function trivial(category_type::Type{<:AbstractCategory})
   return error("`trivial` not defined for type $(category_type).")
 end
 
 istrivial(c::AbstractCategory) = (c == trivial(typeof(c)))
 
-function dual(category_type::Type{<:AbstractCategory})
+# name conflict with LabelledNumber.label. TBD is that an issue?
+label(c::AbstractCategory) = error("method `label` not defined for type $(typeof(c))")
+
+# TBD dimension in Sectors or in GradedAxes namespace?
+function dimension(c::AbstractCategory)
+  return error("method `dimension` not defined for type $(typeof(c))")
+end
+
+function dimension(g::GradedAxes.GradedUnitRange)
```

Comment:
This seems like an example where we could just make the type constraint `g::AbstractUnitRange`, and rely on the object implementing `blocklengths`, and additionally expect that the result of that has an associated label.